### DISCUSSION TITLE
RFC: any chance of teaming up with vishvananda/netlink?

### DISCUSSION THREAD

**Discussion by thediveo** [florianl/go-tc#92] [General]
My background are networking diagnosis tools that need beside the tc aspect of NETLINK also many other NETLINK areas, not least the link-related API functions. Is there any chance to maybe team up with https://github.com/vishvananda/netlink and fill in the blanks when it comes to tc functionality?

I'm not affiliated with the https://github.com/vishvananda/netlink project, just evaluating how to deal with the current modules situation.

**Comment by florianl** [comment]
I'm not sure what you mean by "[...] team up with [vishvananda/netlink](https://github.com/vishvananda/netlink)".

This project focuses purely on the TC part of the netlink family. And like other focused APIs it builds an [ecosystem](https://github.com/mdlayher/netlink#ecosystem) ontop of [the same netlink implementation](https://github.com/mdlayher/netlink).

**Reply by thediveo** [reply]
Thank you very much for this bigger picture, I wasn't aware of it, having come from the (RT)NETLINK vishvananda module.
